File: tests/detectors/samples/S07/python/negative.py
def sync_all():
    page = load_checkpoint()
    while True:
        body = get_page(page)
        for item in body["items"]:
            db.upsert(item)
        save_checkpoint(page + 1)
        if not body["has_more"]:
            break
        page += 1

File: tests/detectors/samples/S07/typescript/negative.ts
export async function syncAll() {
  let page = await loadCheckpoint();
  while (true) {
    const body = await getPage(page);
    for (const item of body.items) {
      await db.upsert({ where: { id: item.id }, create: item, update: item });
    }
    await saveCheckpoint(page + 1);
    if (!body.has_more) break;
    page += 1;
  }
}
